### ISSUE TITLE
setup.py  on  master branch corrupt ?

### ISSUE THREAD

**Issue by anilpai** [jpype-project/jpype#366]
I am trying to create a release and after creating a new release, I do `pip3 install JPype1` and I get the following error.

```
setup.py", line 4, in <module>
        import setupext
    ModuleNotFoundError: No module named 'setupext'
 
```

**Comment by marscher** [comment]
On 30.10.18 22:30, Anil Pai wrote:
> I am trying to create a release and after creating a new release, I do `pip3 install JPype1` and I get the following error.
> 
> ```
> setup.py", line 4, in <module>
>         import setupext
>     ModuleNotFoundError: No module named 'setupext'
>  
> ```
> 

it seems setupext has not been added to master, but resides only in devel.

**Comment by Thrameos** [comment]
It appears to be there to me.  

https://github.com/jpype-project/jpype/tree/master/setupext

**Comment by Thrameos** [comment]
Is it something missing from MANIFEST.in?  I don't know python packaging tool well enough to know how to change what ships with pip.

**Comment by Thrameos** [comment]
I did not replicate the issue.

```
$ python3 -m pip install --force-reinstall JPype1
Collecting JPype1
  Downloading https://files.pythonhosted.org/packages/c4/4b/60a3e63d51714d4d7ef1b1efdf84315d118a0a80a5b085bb52a7e2428cdc/JPype1-0.6.3.tar.gz (168kB)
    100% |████████████████████████████████| 174kB 605kB/s
Building wheels for collected packages: JPype1
  Running setup.py bdist_wheel for JPype1 ... done
  Stored in directory: /home/nelson85/.cache/pip/wheels/0e/2b/e8/c0b818ac4b3d35104d35e48cdc7afe27fc06ea277feed2831a
Successfully built JPype1
Installing collected packages: JPype1
  Found existing installation: JPype1 0.6.3
    Uninstalling JPype1-0.6.3:
      Successfully uninstalled JPype1-0.6.3
Successfully installed JPype1-0.6.3
```

**Comment by anilpai** [comment]
I created a new release and gave that a different name..

```
$ python3 -m pip install --force-reinstall ha_JPype1
Collecting ha_JPype1
  Downloading https://files.pythonhosted.org/packages/38/9c/04d462192b8e897a3f50581f5ae284f9047631cc7a65c9b4e0e22f202e41/ha_JPype1-1.6.7.tar.gz (168kB)
    100% |████████████████████████████████| 174kB 2.6MB/s 
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/7j/ty_pggwd15lcml_pnw3ghmq00000gn/T/pip-build-05z8v158/ha-JPype1/setup.py", line 4, in <module>
        import setupext
    ModuleNotFoundError: No module named 'setupext'

```

**Comment by marscher** [comment]
On 31.10.18 17:53, Anil Pai wrote:
> I created a new release and gave that a different name..
> 
> ```
> $ python3 -m pip install --force-reinstall ha_JPype1
> Collecting ha_JPype1
>   Downloading https://files.pythonhosted.org/packages/38/9c/04d462192b8e897a3f50581f5ae284f9047631cc7a65c9b4e0e22f202e41/ha_JPype1-1.6.7.tar.gz (168kB)
>     100% |████████████████████████████████| 174kB 2.6MB/s 
>     Complete output from command python setup.py egg_info:
>     Traceback (most recent call last):
>       File "<string>", line 1, in <module>
>       File "/private/var/folders/7j/ty_pggwd15lcml_pnw3ghmq00000gn/T/pip-build-05z8v158/ha-JPype1/setup.py", line 4, in <module>
>         import setupext
>     ModuleNotFoundError: No module named 'setupext'
> 
> ```
> 
So you package some non working version and then you blame us for it?

**Comment by Thrameos** [comment]
No I checked.  It gave me a version without setupext from a download site. It appears like python setup uses the same install directory.  So if you install successfully once it finds resources, but not if you pip.  Must be a missing line in the dist building script.

**Comment by anilpai** [comment]
I see a lot of changes to setup file recently.

https://github.com/jpype-project/jpype/commit/490eb6511a0a0b57be2191cbbf3a90bac45e5879#diff-2eeaed663bd0d25b7e608891384b7298L9

I am trying to understand what that means..

**Comment by anilpai** [comment]
Can I also know how to generate a distribution ? how do you do that ? are you guys using twine ?

Any code or instructions will help. Thanks !

**Comment by Thrameos** [comment]
It appears the command we use to build the source distribution ("python setup.py sdist") had a missing line in the manifest file.  I put in a pull request to fix it #367.

**Comment by marscher** [comment]
@anilpai you are aware that you can also install from an git archive, or commit id from github itself. If you upload non-working versions on PyPI some users will stumble upon it and finally blame us for something we never would have released.

You can just use 
```
pip install https://github.com/jpype-project/jpype/archive/161541e229b311f99f75c301404324d3e72a1d49.zip
```
to install a specific version. Why do you need to "release" something in the first place?
